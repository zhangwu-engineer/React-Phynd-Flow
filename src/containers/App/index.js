import 'babel-polyfill';
import 'services/immutability'
import React from 'react';
import { Provider } from 'react-redux'
import { HashRouter } from 'react-router-dom'
import { PersistGate } from 'redux-persist/integration/react'
import { MuiThemeProvider } from '@material-ui/core/styles'
import { ThemeProvider } from 'styled-components'
import { store, persistor, history } from 'services/redux'
import Routes from 'Routes'
import { muiTheme, styledTheme } from 'services/bootstrap';

export const AppContainer = () => (
  <MuiThemeProvider theme={muiTheme}>
    <ThemeProvider theme={styledTheme}>
      <HashRouter basename='/' history={history}>
        <Routes history={history} />
      </HashRouter>
    </ThemeProvider>
  </MuiThemeProvider>
)

export const App = () => (
  <Provider store={store}>
    <PersistGate loading="null" persistor={persistor}>
      <AppContainer />
    </PersistGate>
  </Provider>
)

