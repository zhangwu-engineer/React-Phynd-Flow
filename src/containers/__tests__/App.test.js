import React from 'react';
import ReactDOM from 'react-dom';
import { Provider } from 'react-redux';
import { PersistGate } from 'redux-persist/integration/react'

import { configure, mount, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import { App } from '../App';

describe('To test the App Component functionality.', () => {
  configure({ adapter: new Adapter() });
  let rootComponent = null;

  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<App />, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it(`should mount and check if Provider component exists.`, () => {
    rootComponent = mount(<App />);
    const isReduxProvider = rootComponent.find(Provider);
    expect(isReduxProvider).toHaveLength(1);
  });

  it(`should mount and check if PersistGate component exis`, () => {
    rootComponent = mount(<App />);
    const isPersistGate = rootComponent.find(PersistGate);
    expect(isPersistGate).toHaveLength(1);
  });

});
