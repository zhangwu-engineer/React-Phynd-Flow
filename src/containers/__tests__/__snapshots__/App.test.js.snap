// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`To test the App Component functionality. renders without crashing 1`] = `ShallowWrapper {}`;
